### PR TITLE
General fixes

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Graphics/Containers/Container.cs [hunk 1]
```diff
@@ -339,7 +339,7 @@ public sealed override void UpdateClock(IFrameBasedClock clock)
         /// </summary>
         protected virtual bool RequiresChildrenUpdate => !IsMaskedAway || !childrenSizeDependencies.IsValid;
 
-        internal sealed override bool UpdateSubTree()
+        public override bool UpdateSubTree()
```
Comment (smoogipooo) [inline osu.Framework/Graphics/Containers/Container.cs]: At the very least this needs to be protected internal. Public is too much. Is there really no other way to go about this without exposing UpdateSubTree?

Comment (peppy) [inline osu.Framework/Graphics/Containers/Container.cs]: if you can find another way i'm all ears.

**Comment by smoogipooo** [comment]
@Tom94 Just want to confirm once more that you're okay with this change. I'm against exposing UpdateSubTree so much like this, my proposal is the following:
image

I'll be leaving merging this in your hands otherwise.

**Comment by peppy** [comment]
Keep in mind we also need to be able to block the initial `UpdateSubTree`.

**Comment by peppy** [comment]
We will also need a way to reset variables at the start of a new update loop, so would have to expose an extra method `UpdateBeforeChildren` at very least

**Comment by peppy** [comment]
Also I'm not sure of the naming of this container, but it's ambiguous at best. Again, would need to be accompanied by good xmldoc.

**Comment by peppy** [comment]
This is the accompanying code which would use the proposed class:

untitled

Feels a bit round-about-y byt I'm neutral.

**Comment by Tom94** [comment]
I don't really like the name `UpdateBeforeChildren`, because `Update` is already before children. I do agree with smoogipooo, though, that making `UpdateSubTree` public isn't good.

One thing that could be done is to not iterate updates until `RequiresExtraUpdates == false`, but until our time is equal or further than the parent time (inside `DetachedUpdateContainer`). That would remove the need for `UpdateBeforeChildren`, and might also remove the need for some other redundant state tracking.

**Comment by smoogipooo** [comment]
Let's go with this for now.

**Pull request merged**